test/UnitTests/MSTest.Analyzers.UnitTests/UseNewerAssertThrowsAnalyzerTests.cs

```diff
@@ -0,0 +1,298 @@
+                    Assert.ThrowsExactly<Exception>(action: () => Console.WriteLine());
+                    Assert.ThrowsExactlyAsync<Exception>(() => Task.CompletedTask);
+                    Assert.ThrowsExactlyAsync<Exception>(action: () => Task.CompletedTask);
+            
+                    // action and message overload
+                    Assert.ThrowsExactly<Exception>(() => Console.WriteLine(), "Message");
+                    Assert.ThrowsExactly<Exception>(action: () => Console.WriteLine(), message: "Message");
+                    Assert.ThrowsExactly<Exception>(message: "Message", action: () => Console.WriteLine());
+                    Assert.ThrowsExactly<Exception>(action: () => Console.WriteLine(), "Message");
+                    Assert.ThrowsExactly<Exception>(() => Console.WriteLine(), message: "Message");
+                    Assert.ThrowsExactlyAsync<Exception>(() => Task.CompletedTask, "Message");
+                    Assert.ThrowsExactlyAsync<Exception>(action: () => Task.CompletedTask, message: "Message");
+                    Assert.ThrowsExactlyAsync<Exception>(message: "Message", action: () => Task.CompletedTask);
+                    Assert.ThrowsExactlyAsync<Exception>(action: () => Task.CompletedTask, "Message");
+                    Assert.ThrowsExactlyAsync<Exception>(() => Task.CompletedTask, message: "Message");
+
+                    // action, message, and parameters overload
+                    Assert.ThrowsExactly<Exception>(() => Console.WriteLine(), "Message", "A", "B", "C");
+                    Assert.ThrowsExactly<Exception>(() => Console.WriteLine(), "Message", messageArgs: new object[] { "A", "B", "C" });
+                    Assert.ThrowsExactlyAsync<Exception>(() => Task.CompletedTask, "Message", "A", "B", "C");
+                    Assert.ThrowsExactlyAsync<Exception>(() => Task.CompletedTask, "Message", messageArgs: new object[] { "A", "B", "C" });
+                }
+            }
+            """;
+
+        await VerifyCS.VerifyCodeFixAsync(code, fixedCode);
+    }
+
+    [TestMethod]
+    public async Task WhenAssertThrowsExceptionFuncOverloadExpressionBody_Diagnostic()
+    {
+        string code = """
+            using System;
+            using System.Threading.Tasks;
+            using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+            [TestClass]
+            public class MyTestClass
+            {
+                [TestMethod]
+                public void MyTestMethod()
+                {
+                    [|Assert.ThrowsException<Exception>(() => 5)|];
+                }
+            }
+            """;
+
+        // NOTE: The discard is needed to avoid CS0201: Only assignment, call, increment, decrement, and new object expressions can be used as a statement
+        // This is because ThrowsException has a Func<object> overload that is being used in the original code.
+        // But ThrowsExactly only has an Action overload.
```

Comment:
Interesting, I'll try to have a look at this specific API and see if there is any need to have it on the newer APIs.